### PR TITLE
create devise user model with model tests, create empty UsersController

### PR DESCRIPTION
**Pull request by jraaaaanger**
This adds the User model and user Controller. This build has model tests

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
```
Comment (jraenorris) [inline app/controllers/users_controller.rb]: Was this something generated for you? Are the empty methods doing something devise-y here?

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+RSpec.describe User, type: :model do
+  it { should have_valid(:first_name).when("Jimmy", "Linc")}
+  it { should_not have_valid(:first_name).when(nil, "")}
+
+  it { should have_valid(:last_name).when("Ranger", "Kupke")}
+  it { should_not have_valid(:last_name).when(nil, "")}
+
+  it { should have_valid(:email).when("example@gmail.com", "erlich@aviato.net")}
+  it { should_not have_valid(:email).when("example", "example@", "exam@gma", "example.com", nil)}
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: I might also test for an empty string here

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  describe "#admin?" do
+    context "when user is a admin" do
+      it "should give back true if user an admin" do
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: 'it should return true if user is an admin' would be a little more explicit

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        user = create(:user, role: "admin")
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: I like seeing tests broken into two/ three stages with a single blank line in between. First section should be the setup (if any), next would be the action (if any), then the expectations. It's an organizational tool that helps your team/ future devs know what is going on in your test more explicitly. For a test this small, you could argue not to do it, but it's a great habit to start getting into!

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        expect(user.admin?).to eq(true)
+      end
+    end
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: White space is free! Put a blank line between all tests

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    context "when user is a member" do
+      it 'should give back false' do
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: If you want to use single quotes throughout, you can change the yml file so hound will stop throwing comments in there. That said, try to stick with one or the other, either using single as the default or double as the default

**Comment by jraenorris** [comment]
That's a lot of code for only one commit! Remember: small commits! Always Be Committing!

First pass is done. There are a lot of hound violations still. Address those and my comments and ping me to do a second pass. Nice job!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (jraaaaanger) [inline app/controllers/users_controller.rb]: No, we made these just so we'll have them when we continue

**Comment by jraaaaanger** [comment]
@jraenorris Ready to go!

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (jraenorris) [inline app/controllers/users_controller.rb]: I don't recommend doing this in most cases. It's making assumptions about future development instead of sticking to the single feature you're working on. It could happen that you pivot and never make those routes (probably not for this project, but if you were working on a live product that could totally happen!), so there's confusing unused code hanging out forever

If you keep it in here, it's your responsibility to make sure all those routes get used and track if they are not. You are the owner of that code. You will need to explain why it's there in a week if it goes unused. Git blame remembers. 😄

File: spec/factories/users.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  sequence(:email) { |n| "bg#{n}@pipedpiper.net" }
+  factory :user do
+    first_name "Bertram"
+    last_name "Guilfoyle"
+    email
+    profile_picture "photo.png"
+    role "member"
+    password "SatanCanada"
+    password_confirmation "SatanCanada"
```
Comment (jraenorris) [inline spec/factories/users.rb]: ^ this is a violation to fix

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  it { should_not have_valid(:email).when("example", "example@", "exam@gma", "example.com", "", nil) }
+
+  describe "#admin?" do
+    context "when user is a admin" do
+      it "should return true if user is an admin" do
+        user = create(:user, role: "admin")
+        expect(user.admin?).to eq(true)
+      end
+    end
+
```
Comment (jraenorris) [inline spec/models/user_spec.rb]: ^ whitespace as in spaces (or a tab), not the full line

**Comment by jraenorris** [comment]
@jraaaaanger Looking good. A couple small things then it should be ready to merge. Ping me when you're ready

File: spec/factories/users.rb [hunk 9]
```diff
@@ -0,0 +1,12 @@
+FactoryGirl.define do
+  sequence(:email) { |n| "bg#{n}@pipedpiper.net" }
+  factory :user do
+    first_name "Bertram"
+    last_name "Guilfoyle"
+    email
+    profile_picture "photo.png"
+    role "member"
+    password "SatanCanada"
```
Comment (jraenorris) [inline spec/factories/users.rb]: Is password confirmation required? Look at your schema. Look into password too. It looks like it might be something similar..

Comment (LmKupke) [inline spec/factories/users.rb]: Password Confirmation is required. For creating a new user it has to have password confirmation even for the Factory.
